doc/manual/contract.md

```diff
@@ -0,0 +1,175 @@
+# Contract Programming
+
+Nit supports contract programming like Eiffel. Contracts can be seen as the materialization of specifications in the code itself. The idea is to define the behavior of a property using conditions which will be evaluated on runtime.
+
+In Nit all the conditions of contract must be expressed using boolean expression in the same way as an assertion. Note that the potential routines that compose the condition of the contract should have no side effects.
+
+## Method/Attribute contract
+
+Methods and attributes (setter) can be associated with two types of contracts, entry contracts (expect) and exit contracts (ensure). Together, they make it possible to define the rules to be respected between the called and the caller. If the caller fulfills the `expect` condition on entry then the caller can assume that the called fulfills the `ensure` condition on exit.
```

Comment:
J'utiliserais plutôt le terme "callee" au lieu de "called" pour désigner la méthode appelée. Dans la phrase:
`Together, they make it possible to define the rules to be respected between the called and the caller`
`Together, they define the rules to be respected between the called and the callee`

Les mots "make it possible" sont superflus.

Reply:
Yes you are right i need to use callee :+1: